set.mm

```diff
@@ -674802,6 +674806,831 @@ fixed reference functional determined by this vector (corresponding to
       TYAYAWNUAUBBCDYBYD $.
   $}
 
+
+  $c PrimRoots $.
+  $( Define the class of primitive roots.  (Contributed by metakunt,
+     25-Apr-2025.) $)
+  cprimroots $a class PrimRoots $.
+
+  ${
+    $d a b k l r $.
+    $( A ` r ` -th primitive root is a root of unity such that the exponent
+       divides ` r ` .  (Contributed by metakunt, 25-Apr-2025.) $)
+    df-primroots $a |- PrimRoots = ( r e. CMnd , k e. NN0 |->
+    [_ ( Base ` r ) / b ]_ { a e. b | ( ( k ( .g ` r ) a ) = ( 0g ` r ) /\
+    A. l e. NN0 ( ( l ( .g ` r ) a ) = ( 0g ` r ) -> k || l ) ) } ) $.
+  $}
+
+  ${
+    $d K b k l r x $.  $d M l x $.  $d R b k l r x $.  $d b k l ph r x $.
+    primrootsval.1 $e |- ( ph -> R e. CMnd ) $.
+    primrootsval.2 $e |- ( ph -> K e. NN0 ) $.
+    primrootsval.3 $e |- .^ = ( .g ` R ) $.
+    $( The value of a primitive root.  (Contributed by metakunt,
+       25-Apr-2025.) $)
+    primrootsval $p |- ( ph -> ( M e. ( R PrimRoots K ) <->
```

Comment:
I think the more accurate label is `isprimroot`, like with https://us.metamath.org/mpeuni/isrhm.html